### PR TITLE
 Fixed OffSet paramets for GetUserDialogsAsync & GetHistoryAsync

### PR DESCRIPTION
**Pull request by mkellerman**
Added missing OffSet parameters.

### CODE REVIEW

**Comment by knocte** [comment]
Thanks for your contribution! it's looking good but I'll do a review highlighting just a few nitpicks

File: TLSharp.Core/TelegramClient.cs [hunk 1]
```diff
@@ -258,11 +258,22 @@ public async Task<Boolean> SendTypingAsync(TLAbsInputPeer peer)
             return await SendRequestAsync<Boolean>(req);
         }
 
-        public async Task<TLAbsDialogs> GetUserDialogsAsync()
+        public async Task<TLAbsDialogs> GetUserDialogsAsync(int offset_date = 0, int offset_id = 0, TLAbsInputPeer offset_peer = null, int limit = 100)
```
Comment (knocte) [inline TLSharp.Core/TelegramClient.cs]: if it's a date, shouldn't the param type be of type Date, and let the GetUserDialogAsync convert the param to int under the hood?

File: TLSharp.Core/TelegramClient.cs [hunk 2]
```diff
@@ -258,11 +258,22 @@ public async Task<Boolean> SendTypingAsync(TLAbsInputPeer peer)
...
         {
-            var peer = new TLInputPeerSelf();
-            return await SendRequestAsync<TLAbsDialogs>(
-                new TLRequestGetDialogs() { OffsetDate = 0, OffsetPeer = peer, Limit = 100 });
+            if (!IsUserAuthorized())
+                throw new InvalidOperationException("Authorize user first!");
+
+            if (offset_peer == null)
```
Comment (knocte) [inline TLSharp.Core/TelegramClient.cs]: we use camelCase notation, not under_score, for parameter names

File: TLSharp.Core/TelegramClient.cs [hunk 3]
```diff
@@ -313,17 +324,23 @@ public async Task SendPingAsync()
             await _sender.SendPingAsync();
         }
 
-        public async Task<TLAbsMessages> GetHistoryAsync(TLAbsInputPeer peer, int offset, int max_id, int limit)
+        /// <summary>
+        /// messages.getHistory#dcbb8260 peer:InputPeer offset_id:int offset_date:date add_offset:int limit:int max_id:int min_id:int hash:int = messages.Messages
```
Comment (knocte) [inline TLSharp.Core/TelegramClient.cs]: this XML doc is a bit strange, either make it human readable or remove it please

File: TLSharp.Core/TelegramClient.cs [hunk 4]
```diff
@@ -313,17 +324,23 @@ public async Task SendPingAsync()
...
             {
                 Peer = peer,
-                AddOffset = offset,
+                OffsetId = offset_id,
+                OffsetDate = offset_date,
+                AddOffset = add_offset,
+                Limit = limit,
                 MaxId = max_id,
-                Limit = limit
+                MinId = min_id
```
Comment (knocte) [inline TLSharp.Core/TelegramClient.cs]: please remove trailing whitespace at the end of this line, thanks!

File: TLSharp.Core/TelegramClient.cs (hunk 3 above)
Comment (mkellerman) [reply TLSharp.Core/TelegramClient.cs]: FYI: SearchUserAsync also has the XML doc summary.

File: TLSharp.Core/TelegramClient.cs (hunk 2 above)
Comment (mkellerman) [reply TLSharp.Core/TelegramClient.cs]: FYI: Please look at original code line 316. it was already using 'max_id' under_score parameter names.

**Comment by mkellerman** [comment]
I don't know how to 'modify' a current PR, so I'm going to submit a new one. <newb>

Sorry for the trouble.

**Comment by knocte** [comment]
> I don't know how to 'modify' a current PR

Just push more commits to the same branch.

**Pull request closed without merging**